stl/src/stacktrace.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <format>
+#include <shared_mutex>
+#include <string>
+
+// clang-format off
+#include <Windows.h>
+#include <dbghelp.h>
+// clang-format on
+
+#pragma comment(lib, "Dbghelp.lib")
+
+namespace {
+    static constexpr std::size_t max_line_size = 2000;

```

Comment:
It is function name

Reply:
2000 characters for function name sounds like an even bigger overkill than 2000 characters for file name

Reply:
![spirit_crash](https://user-images.githubusercontent.com/3995422/150833395-f1841578-f62c-4d57-b919-c8ed9ee2a1c0.png)

Reply:
Oh wow, this call stack looks absolutely terrifying

Reply:
Sorry if this question is dumb. I assume that the compiler throws away information about type aliases so you end up with hideous stuff like the above. Has there ever been proposals to keep this information so that errors and stack traces can be more like what the developer typed?

Reply:
It was me doing post-mortem debugging having only a minidump. The crash was in some code that used `boost::spirit` back in 2009. Surprisingly I've managed to understand the issue after switching to WinDbg. 

The function names you see are normal function names for `boost::spirit`. Not sure if 2000 chars is enough.

Reply:
@davidhunter22 , `std::basic_stacktrace<std::allocator<std::stacktrace_entry> >::current` is what I get currently. 

Even if we decide to parse out this to just `current`, the buffer is for full name, not trimmed.

Reply:
And I don't think the implementation should do too smart processing, TBH

Reply:
What enhancement I'm thinking of is adding parameters list. But I'm not sure whether it worth doing, it would add complexity.

Reply:
As for the 2000 limitation, I'd like to avoid any limits at all (up to throwing bad alloc for function name of few gigabytes), but don't see a way to.

Reply:
That is quite something. I'd be interested in a word cloud of that 😅.

Reply:
Ok, instead of arbitrary `static constexpr std::size_t max_line_size = 2000;` will use `MAX_SYM_NAME` defined as 2000 in `DbgHelp.h`